app/experimenter/nimbus-ui/src/hooks/useCommonForm/useCommonNestedForm.tsx

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import { useFormContext } from "react-hook-form";
+import { RegisterOptions, UseFormMethods } from "react-hook-form";
+import { useCommonFormMethods } from "./useCommonFormMethods";
+
+export function useCommonNestedForm<FieldNames extends string>(
+  defaultValues: Record<string, any>,
+  setSubmitErrors: React.Dispatch<React.SetStateAction<Record<string, any>>>,
+  prefix: string,
+  nestedSubmitErrors: Record<string, string[]>,
+  nestedErrors: UseFormMethods["errors"],
+  nestedTouched: UseFormMethods["formState"]["touched"],
+) {
+  const { register, watch } = useFormContext();
```

Comment:
Yeah, this looks pretty clean too... glad we could lean on the react-hook-form context provider and pull this into the wrapper hook